Libraries/vendor/core/_wrapObjectFreezeAndFriends.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @author Ben Newman (@benjamn) <ben@benjamn.com>
+ */
+
+'use strict';
+
+let testMap; // Initialized lazily.
+function getTestMap() {
+  return testMap || (testMap = new (require('./Map'))());
+}
+
+// Wrap Object.{freeze,seal,preventExtensions} so each function adds its
+// argument to a Map first, which gives our ./Map.js polyfill a chance to
+// tag the object before it becomes non-extensible.
+["freeze", "seal", "preventExtensions"].forEach(name => {
+  const method = Object[name];
+  if (typeof method === "function") {
+    Object[name] = function (obj) {
+      try {
+        // If .set succeeds, also call .delete to avoid leaking memory.
+        getTestMap().set(obj, obj).delete(obj);
+      } finally {
+        // If .set fails, the exception will be silently swallowed
+        // by this return-from-finally statement, and the method will
+        // behave exactly as it did before it was wrapped.
```

Comment:
... with the potential of leaking memory, unreported, if the error happens in the `.delete()` portion of things.

Hypothetically; if something wants to call freeze or seal on an object, how likely is it that it actually wouldn't want the `__MAP_POLYFILL_INTERNAL_HASH__` expando applied to the object itself, in the first place? Wouldn't we be better and safer off letting it use the (new-in-this-PR) fallback method instead?

Reply:
> Wouldn't we be better and safer off letting it use the (new-in-this-PR) fallback method instead?

No, because the fallback method has linear lookup time, compared to the tagging approach, which has constant lookup time. That's a nice advantage of this polyfill over other (better tested, more widely used) polyfills that settle for linear time in order to avoid tagging objects. If we don't care about the performance benefit of tagging, then we should really replace this whole polyfill with one that wasn't invented here.

Reply:
I'm confident that `.delete(obj)` will always succeed immediately after `.set(obj, obj)` succeeds in putting `obj` into the map. If that's not the case, there's a bug, and we will fix it.